tests/lib/IframeApi.spec.js

```diff
@@ -0,0 +1,96 @@
+
+    beforeEach(() => {
+        spyOn(CookieJar, 'eat').and.callThrough();
+        spyOn(AccountStore.instance, 'list').and.callThrough();
+        spyOn(KeyStore.instance, 'list').and.callThrough();
+        spyOn(KeyStore.instance, 'doMigrateAccountsToKeys').and.callThrough();
+        spyOnProperty(document, 'cookie', 'get').and.returnValue(Dummy.cookie);
+    });
+
+    it('can list deprecated accounts from cookies on iOS.', async () => {
+        spyOn(BrowserDetection, 'isIos').and.returnValue(true);
+
+        const listedAccounts = await iframeApi.list(true);
+
+        expect(CookieJar.eat).toHaveBeenCalledWith(true);
+        expect(AccountStore.instance.list).not.toHaveBeenCalled();
+        expect(KeyStore.instance.list).not.toHaveBeenCalled();
+        expect(listedAccounts).toEqual(Dummy.deprecatedAccountInfo);
+    });
+
+    it('can list key info from cookies on iOS', async () => {
+        spyOn(BrowserDetection, 'isIos').and.returnValue(true);
+
+        const listedKeys = await iframeApi.list();
+
+        expect(CookieJar.eat).toHaveBeenCalledWith(undefined);
+        expect(AccountStore.instance.list).not.toHaveBeenCalled();
+        expect(KeyStore.instance.list).not.toHaveBeenCalled();
+        expect(listedKeys).toEqual(Dummy.keyInfo);
+    });
+
+    it('can list deprecated accounts from AccountStore on non-iOS', async () => {
+        await Dummy.Utils.createDummyAccountStore();
+        spyOn(BrowserDetection, 'isIos').and.returnValue(false);
+
+        const listedAccounts = await iframeApi.list(true);
+
+        expect(CookieJar.eat).not.toHaveBeenCalled();
+        expect(AccountStore.instance.list).toHaveBeenCalled();
+        expect(KeyStore.instance.list).not.toHaveBeenCalled();
+        expect(listedAccounts).toEqual(Dummy.deprecatedAccountInfo);
+
+        await Dummy.Utils.deleteDummyAccountStore();
+    });
+
+    it('can list key info from KeyStore on non-iOS', async () => {
+        await Dummy.Utils.createDummyKeyStore();
+        spyOn(BrowserDetection, 'isIos').and.returnValue(false);
+
+        const listedKeys = await iframeApi.list();
+
+        expect(CookieJar.eat).not.toHaveBeenCalled();
+        expect(AccountStore.instance.list).not.toHaveBeenCalled();
+        expect(KeyStore.instance.list).toHaveBeenCalled();
+        expect(listedKeys).toEqual(Dummy.keyInfo);
+
+        await Dummy.Utils.deleteDummyKeyStore();
+    });
+
+    it('can migrate by setting the migration flag on iOS', async () => {
+        let cookieSet = false;
+        spyOn(BrowserDetection, 'isIos').and.returnValue(true);
+        spyOnProperty(document, 'cookie', 'set').and.callFake((/** @type {string} */ cookie) => {
+            expect(cookie.startsWith('migrate=1;')).toBe(true);
+            cookieSet = true;
+        });
+
+        await iframeApi.migrateAccountsToKeys();
+
+        expect(cookieSet).toBe(true);
+        expect(KeyStore.instance.doMigrateAccountsToKeys).not.toHaveBeenCalled();
```

Comment:
the cookie kicks off migration once the popup is opended again. what happens if the other cookie is used in the meantime to list the accounts? it will be old accounts until the upgrade happens, right? if the popup wont be opened, the list in the cookie will never be updated, or? 
If yes, then maybe should open the popup to migrate on iOS devices.

Reply:
You are right, we might have to delete the accounts cookie already when setting the migrate signal.

The thing is: this whole thing depends on the iframe being able to write a cookie, that the popup can then read. If that is not possible on iOS, the  we have to use a popup anyway.

Reply:
It's okay to use the old accounts cookie until the accounts get migrated in a popup as the accounts cookie still is up to date as everything that changes the accounts would require a popup anyways